### PR TITLE
Added MapState-dependent bitmap filtering configuration

### PR DESCRIPTION
**Pull request by Nohus**
Implements #17

### CODE REVIEW

File: mapcompose/src/main/java/ovh/plrapps/mapcompose/ui/view/TileCanvas.kt [hunk 1]
```diff
@@ -27,7 +28,8 @@ internal fun TileCanvas(
     alphaTick: Float,
     colorFilterProvider: ColorFilterProvider?,
     tilesToRender: List<Tile>,
-    isBitmapFilteringEnabled: Boolean
+    bitmapFilteringPredicate: (MapState) -> Boolean,
+    mapState: MapState
```
Comment (peterLaurence) [inline mapcompose/src/main/java/ovh/plrapps/mapcompose/ui/view/TileCanvas.kt:32]: The `TileCanvas` doesn't need to depend on `MapState`. Change `bitmapFilteringPredicate` type to `() -> Boolean` and capture de `MapState` in a lambda passed by the parent.
Remove `mapState` parameter

File: mapcompose/src/main/java/ovh/plrapps/mapcompose/ui/MapUI.kt [hunk 2]
```diff
@@ -35,7 +35,8 @@ fun MapUI(
             alphaTick = state.tileCanvasState.alphaTick,
             colorFilterProvider = state.tileCanvasState.colorFilterProvider,
             tilesToRender = state.tileCanvasState.tilesToRender,
-            isBitmapFilteringEnabled = state.isBitmapFilteringEnabled
+            bitmapFilteringPredicate = state.bitmapFilteringPredicate,
```
Comment (peterLaurence) [inline mapcompose/src/main/java/ovh/plrapps/mapcompose/ui/MapUI.kt:38]: `bitmapFilteringPredicate = { bitmapFilteringPredicate(state) }`

File: mapcompose/src/main/java/ovh/plrapps/mapcompose/ui/view/TileCanvas.kt (hunk 1 above)
Comment (peterLaurence) [reply mapcompose/src/main/java/ovh/plrapps/mapcompose/ui/view/TileCanvas.kt:32]: Also, rename `bitmapFilteringPredicate` to `isFilteringBitmap`.

Comment (Nohus) [reply mapcompose/src/main/java/ovh/plrapps/mapcompose/ui/view/TileCanvas.kt]: > The TileCanvas doesn't need to depend on MapState.

Which is why I originally wanted it to only required the scale, not the entire `MapState`, but that's a neat solution, done.

**Pull request merged**